src/core/CMakeLists.txt

```diff
@@ -579,6 +579,12 @@ INCLUDE_DIRECTORIES(
   ${EXPAT_INCLUDE_DIR}
 )
 
+IF(USE_NATIVE_LIB)
+  IF(APPLE)
```

Comment:
Just wondering... Does the core library need to know about mac specific includes?
I assume if there's a base class interface defined with empty method bodies which can be overwritten in platform specific subclasses, all this logic could be kept separate in the src/native folder